src/plugins/embeddable/docs/input_and_output_state.md

```diff
@@ -0,0 +1,237 @@
+## Input and output state
+
+### What's the difference?
+
+Input vs Output State
+
+| Input          | Output |
+| -----------     | ----------- |
+| Public, on the IEmbeddable interface. `embeddable.updateInput(changedInput)`   | Protected inside the Embeddable class. `this.updateOutput(changedOutput)`      |
+|  Serializable representation of the embeddable  | Does not need to be serializable                 |
+|  Can be updated throughout the lifecycle of an Embeddable  | Often derived from input state        |
+
+Non-real examples to showcase the difference:
+
+| Input          | Output      |
+| -----------    | ----------- |
+| savedObjectId  |     savedObjectAttributes  |
+| esQueryRequest |     esQueryResponse        |
+|   props        |     renderComplete         |
+
+### Types of input state
+
+**Inherited input state**
+
+The only reason we have different types of input state is to support embeddable containers, and children embeddables _inheriting_ state from the container. 
+For example, when the dashboard time range changes, so does
+the time range of all children embeddables. Dashboard passes down time range as _inherited_ input state. From the viewpoint of the child Embeddable,
+time range is just input state. It doesn't care where it gets this data from.  
+
+
+For example, imagine a container with this input:
+
+```json
```

Comment:
for some reason these samples have weird rendering: 
![Screenshot 2020-06-18 at 11 03 16](https://user-images.githubusercontent.com/7784120/85000962-613cd500-b153-11ea-9e6a-0e90b54bc3e9.png)
probably because it isn't valid `json`